packages/oauth-client/src/CognitoClient.ts

```diff
@@ -0,0 +1,145 @@
+  return new CognitoError(
+    'UnexpectedResponse',
+    `Unexpected response: ${httpError.response.body}`
+  );
+};
+
+/**
+ * A client for the Cognito API
+ */
+export class CognitoClient extends OAuthClient {
+  readonly clientId: string;
+  readonly clientPassword: string;
+  readonly cognitoLoginUrl: string;
+  readonly redirectUri: string;
+
+  /**
+   * @param {Object} params
+   * @param {string} params.clientId - see example
+   * @param {string} params.clientPassword - see example
+   * @param {string} params.cognitoLoginUrl - see example
+   * @param {string} params.redirectUri - see example
+   *
+   * @example
+   *
+   * const oAuth2Provider = new CognitoClient({
+   *   clientId: 'my-client-id',
+   *   clientPassword: 'my-client-password',
+   *   cognitoLoginUrl: 'https://earthdata.login.nasa.gov',
+   *   redirectUri: 'http://my-api.com'
+   * });
+   */
+
+  constructor(
+    params: {
+      clientId: string,
+      clientPassword: string,
+      cognitoLoginUrl: string,
+      redirectUri: string
+    }
+  ) {
+    if (!params.clientId) throw new TypeError('clientId is required');
+    if (!params.clientPassword) throw new TypeError('clientPassword is required');
+    if (!params.cognitoLoginUrl) throw new TypeError('cognitoLoginUrl is required');
+    if (!params.redirectUri) throw new TypeError('redirectUri is required');
+
+    super({
+      clientId: params.clientId,
+      clientPassword: params.clientPassword,
+      loginUrl: params.cognitoLoginUrl,
+      redirectUri: params.redirectUri,
+    });
+
+    this.clientId = params.clientId;
+    this.clientPassword = params.clientPassword;
+    validateUrl(params.cognitoLoginUrl);
+    this.cognitoLoginUrl = params.cognitoLoginUrl;
+    validateUrl(params.redirectUri);
+    this.redirectUri = params.redirectUri;
+  }
+
+  async getUserInfo(accessToken: string) {
+    if (!accessToken) throw new TypeError('accessToken is required');
+
+    try {
+      const response = await super.getRequest({
+        path: 'oauth/userInfo',
+        accessToken,
+      });
+
+      return response.body;
+    } catch (error) {
+      if (error instanceof got.ParseError) {
+        throw new CognitoError(
+          'InvalidResponse',
+          'Response from Cognito was not valid JSON'
+        );
+      }
+
+      throw httpErrorToCognitoError(error);
+    }
+  }
+
+  /**
+   * Given an authorization code, request an access token and associated
+   * information from the Cognito login service. This overrides the
+   * base class for better, Cognito-specific errors.
+   *
+   * See OAuthClient.getAccessToken(authorizationCode).
+   *
+   * @param {string} authorizationCode - an OAuth2 authorization code
+   * @returns {Promise<Object>} access token information
+   */
+  async getAccessToken(authorizationCode: string): Promise<Object> {
+    try {
+      return await super.getAccessToken(authorizationCode);
+    } catch (error) {
+      throw httpErrorToCognitoError(error);
+    }
+  }
```

Comment:
Both clients have a couple functions overriding the parent class. The only reason I'm doing this is to preserve the difference in error handling between the EDL client and the Cognito client. 

Open to discussion here. If we made the EDL client and the Cognito client handle errors in the same way we could throw a generic `OAuthError` or something from the parent class and remove these overrides.

Reply:
Instead of add the override method to the children class, in the super class, we can do:

async getAccessToken(authorizationCode: string): Promise<Object> {
    try {
    const response = await this.requestAccessToken(authorizationCode);
  catch {
    throw httpErrorToAuthError(error);
  }

And add `httpErrorToAuthError` to both super and children classes.

Reply:
This would do in parent:
  httpErrorToAuthError(error) {
    this.httpErrorToAuthError(error);
  }

Reply:
Actually, should be throw **this.**httpErrorToAuthError(error);
`this` will refer to children's method

Reply:
This is a great idea. Much cleaner.

I moved `httpErrorToAuthError` to a class a method on `OAuthClient`, `CognitoClient`, and `EarthdataLoginClient`. The Cognito and EDL clients no longer have the methods overriding the parent class. 

We could take this further but see what you think of these changes:
https://github.com/nasa/cumulus/pull/2209/commits/95794893c9faf1488569fedd4f3dc702116fd453